test/azul_test_case.py

```diff
@@ -1,13 +1,54 @@
+import logging
 import os
+from typing import List
 from unittest import TestCase
 from unittest.mock import (
     patch,
 )
+import warnings
 
 import boto3.session
 from botocore.credentials import Credentials
 import botocore.session
 
+log = logging.getLogger(__name__)
+
+
+class AzulTestCase(TestCase):
+    _warning_state = None
+    _warning_log = List[warnings.WarningMessage]
+    metadata_api_warning_messages = [
+        "ProjectContact.contact_name is deprecated",
+        "File.file_format is deprecated",
+        "ProjectPublication.publication_title is deprecated",
+        "ProjectPublication.publication_url is deprecated",
+        "CellLine.cell_line_type is deprecated",
+        ".*humancellatlas.data.metadata.api.DissociationProcess",
+        ".*humancellatlas.data.metadata.api.EnrichmentProcess",
+        ".+humancellatlas.data.metadata.api.LibraryPreparationProcess",
+        ".*humancellatlas.data.metadata.api.SequencingProcess"
+    ]
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls._warning_state = warnings.catch_warnings(record=True)
+        cls._warning_log = cls._warning_state.__enter__()
+        warnings.filterwarnings('ignore', '.*<ssl.SSLSocket.*>', ResourceWarning)
+        warnings.filterwarnings('ignore', '.*<socket.socket.*>', ResourceWarning)
+        warnings.filterwarnings('ignore', '.*Call to deprecated method fetch_bundle_manifest.*', DeprecationWarning)
+        for message_ in cls.metadata_api_warning_messages:
+            warnings.filterwarnings('ignore', message_, DeprecationWarning)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        if len(cls._warning_log) > 0:
```

Comment:
```suggestion
        if cls._warning_log:
```

But really just `self.assertFalse(cls._warning_log)`. The backtrace will contain all the unexpected warnings.

Reply:
@hannes-ucsc
I removed the length inspection.

 Would there be other modifications required to use the `self` reference in a `classmethod`?

I like the idea of relying on the assertFalse perform the inspection and the output of the warnings. I'm a little bit confused however; wouldn't assertFalse require the context of a test to be running?